### PR TITLE
Adding github actions

### PR DESCRIPTION
**Pull request by norisalsaadie**

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (giorgiosironi) [inline .github/workflows/ci.yml]: This job should always run on every commit, not just on tags

Comment (norisalsaadie) [reply .github/workflows/ci.yml]: Pushed the wrong file by accident. Pushing the correct one now.

**Comment by NuclearRedeye** [comment]
As a followup once this is merged, we should review our options for restricting repo's to only run actions that have been merged. For example to prevent people from pushing new pull requests with new actions that can then run immediately. I think there are some settings that can control this in the repo settings, but I've not had a chance to play with them myself.

**Comment by norisalsaadie** [comment]
> As a followup once this is merged, we should review our options for restricting repo's to only run actions that have been merged. For example to prevent people from pushing new pull requests with new actions that can then run immediately. I think there are some settings that can control this in the repo settings, but I've not had a chance to play with them myself.

actions_permissions

This seems to be the only permission/restriction section I could find within settings. I'm not sure this achieves the desired restriction. Any thoughts?

**Comment by NuclearRedeye** [comment]
> > As a followup once this is merged, we should review our options for restricting repo's to only run actions that have been merged. For example to prevent people from pushing new pull requests with new actions that can then run immediately. I think there are some settings that can control this in the repo settings, but I've not had a chance to play with them myself.
> 
> actions_permissions
> 
> This seems to be the only permission/restriction section I could find within settings. I'm not sure this achieves the desired restriction. Any thoughts?

I looks like maybe the second option 'might' be a candidate, but it also implies from the text that it won't run actions from other repos which could mean the yarn stuff wouldn't run anymore. I think we'd need to experiment a bit and see how the above work, which is definitely outside of the scope of this issue.
There is another issue somewhere covering a 'potential' migration to GitHub actions, i'll update that instead!

**Comment by diversemix** [comment]
> As a followup once this is merged, we should review our options for restricting repo's to only run actions that have been merged. For example to prevent people from pushing new pull requests with new actions that can then run immediately. I think there are some settings that can control this in the repo settings, but I've not had a chance to play with them myself.

This is a very good point - and goes hand-in-hand with maybe having a "Check your repo settings guide" for Libero?

**Comment by giorgiosironi** [comment]
> For example to prevent people from pushing new pull requests with new actions that can then run immediately

We work from forks in Publisher repositories. As far as I know, pull requests from forks (meaning unauthorized people) run Github Actions but [do not see secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets#using-encrypted-secrets-in-a-workflow) like in Travis CI. So what is the problem of running new actions on an non-trusted pull request?

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]: seems this is exactly how we pass secrets https://github.com/libero/reviewer-client/blob/master/.github/workflows/ci.yml#L98-L100

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]: there's no `build` job listed?

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]: Would like to only execute this step if it is a tag; that way only one workflow definition is needed and we can expand the `on` to be both branches and tags. Checking what kind of `if` is needed here.

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]:
```suggestion
    needs: test
```

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]:
```suggestion
      - uses: borales/actions-yarn@v2.0.0
        if: startsWith(github.ref, 'refs/tags/') 
```

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]: use `name: ` in all `borales/actions-yarn@v2.0.0` steps, otherwise they are just named `Run borales/actions-yarn@v2.0.0` in [the output](https://github.com/libero/eslint-config/runs/410522936?check_suite_focus=true)

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]:
```suggestion
name: CI
```

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]:
```suggestion
  push:
      branches:
      - '*'
```

File: .github/workflows/ci-test.yml
Comment (giorgiosironi) [inline .github/workflows/ci-test.yml]: delete file in favor of unifying in the other workflow and execute the tests only once

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]: rename to `ci.yml` though

File: .github/workflows/ci-publish.yml
Comment (giorgiosironi) [inline .github/workflows/ci-publish.yml]: may need a rename if it doesn't always publish, just build

File: .github/workflows/ci-publish.yml
Comment (norisalsaadie) [reply .github/workflows/ci-publish.yml]: Done

File: .github/workflows/ci-publish.yml
Comment (norisalsaadie) [reply .github/workflows/ci-publish.yml]: done

File: .github/workflows/ci-test.yml
Comment (norisalsaadie) [reply .github/workflows/ci-test.yml]: Not sure this is a good idea. Having an action only trigger on the right conditions seems better to me. You are unlikely to have a run away event where you end up with 5 or more actions.

File: .github/workflows/ci.yml
Comment (giorgiosironi) [inline .github/workflows/ci.yml:5]: does this cover tags or is `tags: - '*'` needed?

File: .github/workflows/ci.yml
Comment (giorgiosironi) [inline .github/workflows/ci.yml]: can this be moved to just the `Yarn publish` step? https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif That way `yarn build` (including TypeScript compilation?) is always executed

File: .github/workflows/ci.yml
Comment (giorgiosironi) [inline .github/workflows/ci.yml:22]: locally `yarn build` fails for me with:
```
$ yarn build
yarn run v1.12.3
$ tsc
/bin/sh: 1: tsc: not found
```
but passes after `yarn install`. AFAIK different jobs are different VMs though, so won't have `node_modules`?

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**
